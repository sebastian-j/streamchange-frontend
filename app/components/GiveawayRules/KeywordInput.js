import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import InputAdornment from '@material-ui/core/InputAdornment';
import Tooltip from '@material-ui/core/Tooltip';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import StyledTextField from '../StyledTextField';

const ToggleVisibilityBtn = styled.button`
  background-color: transparent;
  border: none;
  color: ${props => props.theme.staticTextColor};
  outline: none;
  &:hover {
    color: #0094ff;
  }
  & * {
    pointer-events: none;
  }
`;

const KeywordInput = () => {
  const [keyword, setKeyword] = useState('');
  const [visible, setVisible] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    let storedKeyword = localStorage.getItem('keyword');
    if (storedKeyword === null) {
      storedKeyword = '';
    }
    setKeyword(storedKeyword);
  }, []);

  const handleInputChange = event => {
    if (event.target.value.includes(localStorage.getItem('gv-abortCommand'))) {
      setError('Komenda na dołączenie nie może zawierać komendy rezygnacji.');
      return;
    }
    setError(null);
    setKeyword(event.target.value);
    localStorage.setItem('keyword', event.target.value);
  };
  const handleToggleButton = () => setVisible(prevState => !prevState);

  return (
    <FormattedMessage {...messages.keyword}>
      {label => (
        <StyledTextField
          autoFocus
          error={!!error}
          margin="dense"
          name="keyword"
          onChange={handleInputChange}
          id="keyword"
          label={label}
          type={visible ? 'text' : 'password'}
          value={keyword}
          helperText={error}
          fullWidth
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <FormattedMessage {...messages.keywordVisibility}>
                  {title => (
                    <Tooltip title={title} aria-label="add">
                      <ToggleVisibilityBtn
                        aria-label="toggle keyword visibility"
                        name="showKeyword"
                        onClick={handleToggleButton}
                        type="button"
                      >
                        {visible ? (
                          <svg width="24" height="24" viewBox="0 0 24 24">
                            <path d="M0 0h24v24H0z" fill="none" />
                            <path d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z" />
                          </svg>
                        ) : (
                          <svg width="24" height="24" viewBox="0 0 24 24">
                            <path
                              d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
                              fill="none"
                            />
                            <path d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z" />
                          </svg>
                        )}
                      </ToggleVisibilityBtn>
                    </Tooltip>
                  )}
                </FormattedMessage>
              </InputAdornment>
            ),
          }}
        />
      )}
    </FormattedMessage>
  );
};

export default KeywordInput;
