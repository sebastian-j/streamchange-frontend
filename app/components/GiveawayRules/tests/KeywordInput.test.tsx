import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import KeywordInput from '../KeywordInput';

const mockStore = configureStore([]);

describe('<KeywordInput />', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      keyword: 'Keyword',
      prize: 'Prize',
    });
  });
  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale="en">
            <KeywordInput />
          </IntlProvider>
        </Provider>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
