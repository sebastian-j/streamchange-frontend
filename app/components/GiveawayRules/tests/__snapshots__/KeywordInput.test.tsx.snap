// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeywordInput /> should render and match the snapshot 1`] = `
<div
  aria-describedby={null}
  aria-label="keyword"
  className="MuiFormControl-root MuiTextField-root sc-bdvvtL gENRTb MuiFormControl-marginDense MuiFormControl-fullWidth"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title={null}
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
    data-shrink={false}
    htmlFor="keyword"
    id="keyword-label"
  >
    Keyword
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd MuiInputBase-marginDense MuiInput-marginDense"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
      disabled={false}
      id="keyword"
      name="keyword"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
    >
      <button
        aria-describedby={null}
        aria-label="toggle keyword visibility"
        className="sc-dkPtRN evjlnr"
        name="showKeyword"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title={null}
        type="button"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
      </button>
    </div>
  </div>
  
</div>
`;
